strawberry/experimental/pydantic/object_type.py

```diff
@@ -25,15 +26,30 @@
 
 from .exceptions import MissingFieldsListError, UnregisteredTypeException
 
+create_type = type
 
-def replace_pydantic_types(type_: Any):
+
+@dataclasses.dataclass
+class RegisterNestedOptions:
+    """Options for registering nested pydantic model"""
+
+    format_type: Callable[[str], str] = lambda x: x # Default does not rename the nested model's name
+    # todo: Other options such as exclude all fields that have a certain name
+
```

Comment:
i didn't put these arguments in the decorator arguments because these options only apply if register_nested is passed. 
format_type is quite useful in my use case because we frequently have to rename "Type" into "Type{ServiceName}".

Alternatively, we could support a bunch of class decorator for pydantic basemodels. which is what the library [Caliban does for scala](https://ghostdogpr.github.io/caliban/docs/schema.html#annotations).

```
@GQLName("UserGQL") # rename its to UserGQL when it gets automatically registered
class User(pydantic.BaseModel):
     ...
```